src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -669,8 +674,8 @@ class PySplitArgs:
     def passthru(self) -> list[str]: ...
 
 class PyArgSplitter:
-    def __init__(self, build_root: str, known_goals: list[str]) -> None: ...
-    def split_args(self, args: list[str]) -> PySplitArgs: ...
+    def __init__(self, build_root: str, known_goals: tuple[PyGoalInfo, ...]) -> None: ...
+    def split_args(self, args: tuple[str, ...]) -> PySplitArgs: ...
```

Comment:
I prefer using `tuple` to `list` here as it hints to the reader that the arg won't be mutated (obviously it won't be because this is just copied on the Rust side, but it's nice to telegraph that to the casual reader)